.github/workflows/model-1-build.yml

```diff
@@ -90,49 +100,38 @@ jobs:
           echo "GH_REF=${{ github.ref_name }}" >> $GITHUB_ENV
         fi
 
-    - name: Activate spack
-      run: . $SPACK_ROOT/share/spack/setup-env.sh
-
-    - name: Activate environment
-      run: spack env activate ${{ needs.setup-model.outputs.package-name }}
-
-    - name: Build package via spack
-      run: spack -d install --add --only package --fail-fast --no-checksum ${{ needs.setup-model.outputs.package-name }}@=$GH_REF%${{ matrix.compiler.name }}@${{ matrix.compiler.version }}
-
-    - name: Generate lockfile
+    - name: Build ${{ env.PACKAGE_NAME }} via spack
       run: |
-        cd $SPACK_ROOT/var/spack/environments/${{ needs.setup-model.outputs.package-name }}
-        cp spack.lock ${{ needs.setup-model.outputs.package-name }}.original.spack.lock
+        . $SPACK_ROOT/share/spack/setup-env.sh
+        spack load ${{ matrix.compiler.package }}@${{ matrix.compiler.version }}
+        spack compiler find
+        spack env activate ${{ env.PACKAGE_NAME }}
```

Comment:
In the quest to have an easy way to test CI locally with the docker images, it would be good if the commands required to reproduce the CI environment were easily reproducible.

To that end, would it make sense to put them in a separate script in the repo? That would require passing it some variables, which might be a pain and so undesirable.

Another option would be to echo the commands necessary to reproduce the environment when the CI fails. That way someone can look at the logs and copy and paste the exact commands required. Not sure if there is a nice way to do that once and reuse it though.

Reply:
In the logs, I believe it substitutes the values noted (like `matrix.compiler.package` and `env.PACKAGE_NAME`).